From e64c30caab0e9efe8636e1bb3702ef3e75339564 Mon Sep 17 00:00:00 2001
From: Vitalii Vorobiov <vi.vorobiov@samsung.com>
Date: Fri, 18 Mar 2016 14:00:51 +0200
Subject: [PATCH 25/79] edje_edit: program's source and signal CAN be NULL

Or else user can't remove them, only by writing source/signal like "" or some
gibberish which is still getting interprettered as source and signal
---
 src/lib/edje/edje_edit.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/lib/edje/edje_edit.c b/src/lib/edje/edje_edit.c
index 3934392..05ed301 100644
--- a/src/lib/edje/edje_edit.c
+++ b/src/lib/edje/edje_edit.c
@@ -9674,8 +9674,6 @@ edje_edit_program_source_set(Evas_Object *obj, const char *prog, const char *sou
    GET_ED_OR_RETURN(EINA_FALSE);
    GET_EPR_OR_RETURN(EINA_FALSE);
 
-   if (!source) return EINA_FALSE;
-
    /* Remove from program array */
    _edje_program_remove(ed->collection, epr);
 
@@ -9848,8 +9846,6 @@ edje_edit_program_signal_set(Evas_Object *obj, const char *prog, const char *sig
    GET_ED_OR_RETURN(EINA_FALSE);
    GET_EPR_OR_RETURN(EINA_FALSE);
 
-   if (!sig) return EINA_FALSE;
-
    /* Remove from program array */
    _edje_program_remove(ed->collection, epr);
 
-- 
2.7.4

